test/apicoverage/tools/main.go

```diff
@@ -44,22 +56,50 @@ func main() {
 		}
 	}
 
+	var buildFailed = os.Getenv("BUILD_FAILED")
+	if buildFailed != "" && buildFailed == "1" {
+		if err := tools.WriteResourcePercentages(path.Join(
+			artifactsDir, "junit_bazel.xml"),
+			getFailedResourceCoverages()); err != nil {
+			log.Fatalf("Failed writing resource coverage percentages: %v",
+				err)
+		} else {
+			return
+		}
+	}
+
 	if kubeConfigPath, err = tools.GetDefaultKubePath(); err != nil {
 		log.Fatalf("Error retrieving kubeConfig path: %v", err)
 	}
 
-	if serviceIP, err = tools.GetWebhookServiceIP(kubeConfigPath, "", common.WebhookNamespace, common.CommonComponentName); err != nil {
+	if serviceIP, err = tools.GetWebhookServiceIP(kubeConfigPath, "",
+		common.WebhookNamespace, common.CommonComponentName); err != nil {
 		log.Fatalf("Error retrieving Service IP: %v", err)
 	}
 
 	for resource := range common.ResourceMap {
-		err = tools.GetAndWriteResourceCoverage(serviceIP, resource.Kind, path.Join(artifactsDir, strings.ToLower(resource.Kind)+".html"), rules.GetDisplayRules())
+		err = tools.GetAndWriteResourceCoverage(serviceIP, resource.Kind,
+			path.Join(artifactsDir, strings.ToLower(resource.Kind)+".html"),
+			rules.GetDisplayRules())
 		if err != nil {
-			log.Println(fmt.Sprintf("resource coverage for resource: %s failed. %v ", resource.Kind, err))
+			log.Println(fmt.Sprintf("Failed retrieving resource coverage for" +
```

Comment:
Why not just `log.Printf` with a trailing `\n` instead of Println+Sprintf?

Reply:
My bad, fixed it with `log.Printf`